enzy_htp/structure/structure_region/capping.py

```diff
@@ -50,124 +62,222 @@ def capping_with_residue_terminals(raw_region,
+        if raw_region.needs_nterm_cap(res):
+            new_atoms.extend(cap_residue_free_terminal(res, nterm_cap, "nterm"))
+
+        new_atoms.extend(res.atoms)
+
+        if raw_region.needs_cterm_cap(res):
+            new_atoms.extend(cap_residue_free_terminal(res, cterm_cap, "cterm"))
+
     if return_copy:
         new_region = raw_region.clone()
-        new_region.atoms.extend(add_atoms)
+        new_region.atoms = new_atoms 
         return new_region
     else:
-        raw_region.atoms.extend(add_atoms)
-
-# endregion
+        raw_region.atoms = new_atoms 
 
-class ResidueCap(Residue):
-    """the residue cap that link to the corresponding
+class ResidueCap(Residue, ABC):
+    """The residue cap that link to the corresponding
     Residue while not having the children atoms actually in
-    that Residue.
+    that Residue. It ensures that given the defined terminal atom, it can 
+    generate a realistic geometry.
+
     This is mainly used in handling cases like capping atoms that
-    does not exists in the real structure.
-    Attribute:
-        link_residue
-        children/atoms
-        parent = None
-        link_atom:
-            the linking atom in original res
-        socket_atom:
-            the socket from original stru to pluging the cap.
-            the cap is replacing the atom.
-        plug_atom:
-            the pluging in atom in the cap
+    do not exist in the real structure.
+
+    Attributes:
+        link_residue: The Residue() being capped.
+        link_atom: THe terminal Atom() that needs capping specifically.
+        socket_atom: Atom() that connects to the link_atom in the normal Structure().
+        _terminal_type: What type of terminal is it? n-term, c-term, etc.
+        atoms: A List[Atom] of the Atom()'s in the ResidueCap.
     
-    Property:
-        cap_type"""
+    Properties:
+        idx: The idx of the Residue() being capped.
+        chain: The chain name of the Residue() being capped.
+        name: The residue name of the Residue() being capped.
+        term_type: The property getter of _terminal_type
+        net_charge: The net charge of the ResidueCap() chemical group.
+        """
     def __init__(self, link_residue: Residue,
                  link_atom: Atom, socket_atom: Atom,
-                 atoms: List[Atom], plug_atom: Atom):
+                    terminal_type:str):
+        """Simplistic ctor. Will throw if link_atom is NOT in link_residue."""
         self.link_residue = link_residue
         self.link_atom = link_atom
         self.socket_atom = socket_atom
         if link_atom not in link_residue.atoms:
             _LOGGER.error("link atom must be one of the link residue atoms")
             raise ValueError
-        self.atoms = atoms
-        self.plug_atom = plug_atom
-        if plug_atom not in self.atoms:
-            _LOGGER.error("plug atom must be one of the atoms")
-            raise ValueError
 
-        self._idx = None
-        self._name = None
```

Comment:
We need to keep this to make logging normal if not working methods from Residues is called

Reply:
Can you elaborate on this? I am confused what you mean.

Reply:
[after meeting]
This is needed by a temp fix from xtb interface for using .mol correctly.